app/models/concerns/price_validator.rb

```diff
@@ -0,0 +1,8 @@
+class PriceValidator < ActiveModel::Validator
+  def validate(record)
+    unless record.price > record.discount_price
```

Comment:
what would happen if record does not have a price

Reply:
or a discount_price

Reply:
done